engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala

```diff
@@ -1,242 +1,33 @@
 package cromwell.engine.workflow.workflowstore
 
-import java.time.OffsetDateTime
-
-import akka.actor.{ActorLogging, ActorRef, LoggingFSM, Props}
+import akka.actor.{Actor, ActorLogging, ActorRef, Props}
 import cats.data.NonEmptyList
 import cromwell.core._
 import cromwell.core.Dispatcher.EngineDispatcher
-import cromwell.engine.workflow.WorkflowManagerActor
-import cromwell.engine.workflow.WorkflowManagerActor.WorkflowNotFoundException
-import cromwell.engine.workflow.workflowstore.WorkflowStoreActor._
-import cromwell.engine.workflow.workflowstore.WorkflowStoreState.StartableState
-import cromwell.services.metadata.MetadataService.{MetadataPutAcknowledgement, PutMetadataAction}
-import cromwell.services.metadata.{MetadataEvent, MetadataKey, MetadataValue}
-import lenthall.util.TryUtil
-import org.apache.commons.lang3.exception.ExceptionUtils
-
-import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success, Try}
-
-case class WorkflowStoreActor(store: WorkflowStore, serviceRegistryActor: ActorRef)
-  extends LoggingFSM[WorkflowStoreActorState, WorkflowStoreActorData] with ActorLogging {
-
-  implicit val ec: ExecutionContext = context.dispatcher
-
-  startWith(Unstarted, WorkflowStoreActorData(None, List.empty))
-  self ! InitializerCommand
 
-  when(Unstarted) {
-    case Event(InitializerCommand, _) =>
-      val work = store.initialize map { _ =>
-        log.debug("Workflow store initialization successful")
-      }
-      addWorkCompletionHooks(InitializerCommand, work)
-      goto(Working) using stateData.withCurrentCommand(InitializerCommand, sender)
-    case Event(x: WorkflowStoreActorCommand, _) =>
-      stay using stateData.withPendingCommand(x, sender)
-  }
-
-  when(Idle) {
-    case Event(cmd: WorkflowStoreActorCommand, _) =>
-      if (stateData.currentOperation.nonEmpty || stateData.pendingOperations.nonEmpty) {
-        log.error("Non-empty WorkflowStoreActorData when in Idle state: {}", stateData)
-      }
-      startNewWork(cmd, sender, stateData.withCurrentCommand(cmd, sender))
-  }
-
-  when(Working) {
-    case Event(WorkDone, data) =>
-      val newData = data.pop
-      newData.currentOperation match {
-        case None => goto(Idle) using newData
-        case Some(WorkflowStoreActorCommandWithSender(cmd, sndr)) => startNewWork(cmd, sndr, newData)
-      }
-    case Event(cmd: WorkflowStoreActorCommand, data) => stay using data.withPendingCommand(cmd, sender)
-  }
+final case class WorkflowStoreActor(store: WorkflowStore, serviceRegistryActor: ActorRef) extends Actor with ActorLogging {
```

Comment:
IMO it's the same concept as an interface where I changed the implementation behind the scenes. I disagree that they're wholly unrelated, together they form what we currently think of as the workflow store, whereas previously that was accomplished by a single point.